### PR TITLE
Add the Spring native support

### PR DESCRIPTION
**Pull request by moarychan**
Add a new section for the Spring Native support introduction, see [Feature req](https://github.com/Azure/azure-sdk-for-java/issues/28053) for more details.

### CODE REVIEW

**Comment by chenrujun** [comment]
Please don't merge this PR before this PR merged: https://github.com/MicrosoftDocs/azure-dev-docs-pr/pull/2923

And merge to MS docs first, because we set MS docs as single source of truth.

cc: @hui1110 , FYI.

File: docs/src/main/asciidoc/spring-native-support.adoc
Comment (saragluna) [inline docs/src/main/asciidoc/spring-native-support.adoc:24]: The `Dependencies` section in our other part, like Spring Data, Spring Integration, is used to show users how to add library into their application.

**Comment by saragluna** [comment]
Discussed the outline offline, and below is the first draft:

1 Spring Native
2 Support
  2.1 Spring Cloud Azure Native
  2.2 Spring Native
  2.3 Limitations
3 Project Setup
4 Build the native application
  4.1 Build with Buildpacks
  4.2 Build with Native Build Tools
5 Run the native application
  5.1 Build with Buildpacks
  5.2 Build with Native Build Tools

File: docs/src/main/asciidoc/spring-native-support.adoc
Comment (saragluna) [inline docs/src/main/asciidoc/spring-native-support.adoc:10]: Perhaps [Spring Native Support] is better than '[Support]'.

File: docs/src/main/asciidoc/spring-native-support.adoc
Comment (saragluna) [inline docs/src/main/asciidoc/spring-native-support.adoc:20]: for Spring Cloud Azure ~~level~~

Comment (saragluna) [reply docs/src/main/asciidoc/spring-native-support.adoc:20]: to build ~~Spring Cloud Azure~~ applications to native executables.

Comment (saragluna) [reply docs/src/main/asciidoc/spring-native-support.adoc:20]: No extra modifications are needed to the code that uses Spring Cloud Azure libraries apart from adding the dependency.

File: docs/src/main/asciidoc/spring-native-support.adoc
Comment (saragluna) [inline docs/src/main/asciidoc/spring-native-support.adoc:24]: clients auto-configuration

File: docs/src/main/asciidoc/spring-native-support.adoc
Comment (saragluna) [inline docs/src/main/asciidoc/spring-native-support.adoc:27]: move spring integration entries to the end of this list

File: docs/src/main/asciidoc/spring-native-support.adoc
Comment (saragluna) [inline docs/src/main/asciidoc/spring-native-support.adoc:45]: move this to #2

File: docs/src/main/asciidoc/spring-native-support.adoc
Comment (saragluna) [inline docs/src/main/asciidoc/spring-native-support.adoc:53]: Cloud

**Review by saragluna — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc
Comment (moarychan) [reply docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:70]: It will enable the reference from the `attributes.adoc`, and the name `_attributes.adoc` can not enable the reference success.

**Review by saragluna — APPROVED** [review]

**Comment by moarychan** [comment]
I will remove the *spring-native-support.adoc* file, and convert it to a markdown file, append it to the Yonghui's PR for MS doc sync.

**Comment by hui1110** [comment]
https://github.com/MicrosoftDocs/azure-dev-docs-pr/pull/3035

**Comment by chenrujun** [comment]
Closing this PR. 
@hui1110 , please update MS docs first, then sync to this repo.

**Pull request closed without merging**
